javaparser-core/src/main/java/com/github/javaparser/ast/stmt/SwitchEntry.java

```diff
@@ -202,6 +217,12 @@ public boolean remove(Node node) {
         if (node == null) {
             return false;
         }
+        if (guard != null) {
```

Comment:
Why this and the counterpart?
Do you have a test case that tests the deletion of a node with a guard expression?